src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -890,6 +909,30 @@ protected virtual async Task OnDeleteBlockAsync(SyncBlockInfo block)
          await Task.CompletedTask;
       }
 
+      public QueryResult<QueryMempoolTransaction> GetMemoryTransactionsSlim(int offset, int limit)
+      {
+         ICollection<MempoolTable> list = Mempool.AsQueryable().Skip(offset).Take(limit).ToList();
```

Comment:
Is the table already sorted correctly?

Reply:
Not sure what to sort on, I could add a datetime property first time the trx was seen by us and we can sort by that (it will have to be indexed)

Reply:
Add a FirstSeen parameter